cstore_reader.c

```diff
@@ -200,6 +214,85 @@ CStoreReadFooter(StringInfo tableFooterFilename)
 }
 
 
+/* DeserializeBlockData  deserializes requested data block for all columns. It
+ * uncompresses serialized data if necessary, sets deserialized buffer index at
+ * read state. The function also deallocates previous data buffer used to keep temporary
+ * uncompressed data buffer.
+ */
+static void
+DeserializeBlockData(TupleDesc tupleDescriptor, StripeData *stripeData,
+					 MemoryContext stripeReadContext, uint64 blockIndex,
+					 BlockDataBuffers **blockDataBuffersArray)
```

Comment:
It is not clear from the signature what is the function's input and what is the output. Maybe we can do something like this:

```C
static void
DeserializeBlockData(StripeDataBuffers *stripeBuffers, uint32 blockIndex,
					 TupleDesc tupleDescriptor,
					 ColumnBlockData **outputBlockDataArray);
```

This passes what's going to be deserialized as the first argument, then the helper variables as the 2nd and 3rd arguments, and then finally the output argument with a distinctive name. Please make sure to use the same ordering and naming style that you have used for other similar functions. (e.g. should we use ```outputBlockDataArray``` or ```resultBlockDataArray```? etc.)